app/plugin.go

```diff
@@ -666,6 +737,10 @@ func (a *App) getPluginsFromFolder() (map[string]*pluginSignaturePaths, *model.A
 	if appErr != nil {
 		return nil, model.NewAppError("getPluginsFromDir", "app.plugin.sync.list_filestore.app_error", nil, appErr.Error(), http.StatusInternalServerError)
 	}
+	return getPluginsFromFilePaths(fileStorePaths), nil
+}
+
+func getPluginsFromFilePaths(fileStorePaths []string) map[string]*pluginSignaturePaths {
```

Comment:
`pluginSignaturePaths` already contains the `id`. Why do I need a map?

Reply:
To search in O(1)

Reply:
Code is simpler this way. You don't need to iterate over plugins to find one.

Reply:
Not going to block on this discussion.